From a77b56f122d7309b67605537d547c0d3d3639a18 Mon Sep 17 00:00:00 2001
From: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Date: Thu, 29 Mar 2018 08:46:49 -0700
Subject: [PATCH] Avoid forcing CMAKE_VERBOSE_MAKEFILE: let user chose
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

CMAKE_VERBOSE_MAKEFILE causes output of the whole build command lines
to the console instead of just file name being compiled or linked. This
greatly obscures the output! As a result we have missed the following
gcc compiler warning as a minimum:
  cc1: warning: command line option ‘-Wnon-virtual-dtor’ is valid for C++/ObjC++ but not for C [enabled by default]

This change removes CMAKE_VERBOSE_MAKEFILE forced to true. User can
either specify it on a cmake confugre command line as:
  mkdir build && cd build
  cmake -DCMAKE_VERBOSE_MAKEFILE=TRUE ..
or simply force output during the build:
  make VERBOSE=1

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
---
 api/opensource/mfx_dispatch/CMakeLists.txt     | 1 -
 builder/FindGlobals.cmake                      | 1 -
 contrib/SafeStringStaticLibrary/CMakeLists.txt | 1 -
 3 files changed, 3 deletions(-)

diff --git a/api/opensource/mfx_dispatch/CMakeLists.txt b/api/opensource/mfx_dispatch/CMakeLists.txt
index 17d23e5..dbe14e9 100644
--- a/api/opensource/mfx_dispatch/CMakeLists.txt
+++ b/api/opensource/mfx_dispatch/CMakeLists.txt
@@ -33,7 +33,6 @@ function( get_api_version mfx_version_major mfx_version_minor )
   set(${mfx_version_minor} ${minor} PARENT_SCOPE)
 endfunction()
 
-set( CMAKE_VERBOSE_MAKEFILE             TRUE )
 set( CMAKE_LIB_DIR ${CMAKE_BINARY_DIR}/__lib )
 
 if( CMAKE_SYSTEM_NAME MATCHES Windows )
diff --git a/builder/FindGlobals.cmake b/builder/FindGlobals.cmake
index 05f190e..2e35ab6 100644
--- a/builder/FindGlobals.cmake
+++ b/builder/FindGlobals.cmake
@@ -19,7 +19,6 @@
 # SOFTWARE.
 
 set_property( GLOBAL PROPERTY USE_FOLDERS ON )
-set( CMAKE_VERBOSE_MAKEFILE             TRUE )
 
 # the following options should disable rpath in both build and install cases
 set( CMAKE_INSTALL_RPATH "" )
diff --git a/contrib/SafeStringStaticLibrary/CMakeLists.txt b/contrib/SafeStringStaticLibrary/CMakeLists.txt
index a1967c5..d6fd47e 100644
--- a/contrib/SafeStringStaticLibrary/CMakeLists.txt
+++ b/contrib/SafeStringStaticLibrary/CMakeLists.txt
@@ -1,7 +1,6 @@
 cmake_minimum_required( VERSION 2.6.2 FATAL_ERROR )
 project( SafeString )
 
-set( CMAKE_VERBOSE_MAKEFILE             TRUE )
 set( CMAKE_LIB_DIR ${CMAKE_BINARY_DIR}/__lib )
 
 if( ${CMAKE_SYSTEM_NAME} MATCHES Windows )
